tests/config.go

```diff
@@ -34,22 +36,35 @@ type Nodes struct {
 }
 
 // NewConfig creates a new config.
-func NewConfig() *Config {
+func NewConfig() (*Config, error) {
 	cfg := &Config{}
 	flag.StringVar(&cfg.configFile, "config", "", "Config file")
 	flag.StringVar(&cfg.LogDir, "log-dir", "/logDir", "log directory")
 	flag.IntVar(&cfg.FaultTriggerPort, "fault-trigger-port", 23332, "the http port of fault trigger service")
 	flag.StringVar(&cfg.TidbVersions, "tidb-versions", "v2.1.3,v2.1.4", "tidb versions")
 	flag.StringVar(&cfg.OperatorTag, "operator-tag", "master", "operator tag used to choose charts")
 	flag.StringVar(&cfg.OperatorImage, "operator-image", "pingcap/tidb-operator:latest", "operator image")
-	flag.StringVar(&cfg.OperatorRepoDir, "operator-repo-dir", "/tidb-operator", "local directory to which tidb-operator cloned")
 	flag.Parse()
 
-	return cfg
+	operatorRepo, err := ioutil.TempDir("", "tidb-operator")
+	if err != nil {
+		return nil, err
+	}
+	cfg.OperatorRepoDir = operatorRepo
+
+	chartDir, err := ioutil.TempDir("", "charts")
+	if err != nil {
+		return nil, err
+	}
+	cfg.ChartDir = chartDir
+	return cfg, nil
```

Comment:
Where is the place to clean these temporary directories?

Reply:
done